### PR TITLE
BB-237 fix(validateDates): Accept dates in YYYY-MM-DD, YYYY-MM and YYYY formats. Also ensure the begin date is less than the end date.

### PR DESCRIPTION
**Pull request by shivam-tripathi**
### Problem

Dates presently are being accepted in  ISO 8601 format which allows for dates being entered in formats different than ones mentioned above.

### Solution
The PR utilises moment, already a dependency in the project to verify date to the exact format, and does away with the validator.isDate method.

### Areas of Impact
src/client/entity-editor/validators/base.js

### CODE REVIEW

**Comment by LordSputnik** [comment]
We also need to accept YYYY and YYYY-MM.

**Comment by shivam-tripathi** [comment]
@LordSputnik, How does it look now? :)

File: src/client/entity-editor/validators/base.js [hunk 1]
```diff
@@ -20,9 +20,12 @@
 
 import {Iterable} from 'immutable';
 import _ from 'lodash';
+import moment from 'moment';
 import validator from 'validator';
 
 
+const dateFormats = ['YYYY-MM-DD', 'YYYY-MM', 'YYYY'];
```
Comment (LordSputnik) [inline src/client/entity-editor/validators/base.js]: A constant called VALID_DATE_FORMATS might be better here (if the linter allows that name).

**Comment by shivam-tripathi** [comment]
Normally, this PR should have failed tests on validate*SectionendDate, but that didn't happen as the test script in the package.json wasn't running all the tests (more precisely, those in the subdirectories). The PR #193 solves this, and I will be adding fixes to the broken tests.

**Comment by shivam-tripathi** [comment]
@LordSputnik Good to go from my side. The test coverage would increase once #193 is merged.

**Comment by MonkeyDo** [comment]
Thanks @shivam-tripathi ! :)
This took a while, thanks for your patience

**Pull request merged**